diff --git base/trace_event/malloc_dump_provider.cc base/trace_event/malloc_dump_provider.cc
index bf6a3f68b6706..359cccfcddff4 100644
--- base/trace_event/malloc_dump_provider.cc
+++ base/trace_event/malloc_dump_provider.cc
@@ -211,8 +211,8 @@ void ReportMallinfoStats(ProcessMemoryDump* pmd,
   // In case of Android's jemalloc |arena| is 0 and the outer pages size is
   // reported by |hblkhd|. In case of dlmalloc the total is given by
   // |arena| + |hblkhd|. For more details see link: http://goo.gl/fMR8lF.
-  *total_virtual_size += checked_cast<size_t>(info.arena + info.hblkhd);
-  size_t total_allocated_size = checked_cast<size_t>(info.uordblks);
+  *total_virtual_size += checked_cast<size_t>((size_t)info.arena + (size_t)info.hblkhd);
+  size_t total_allocated_size = checked_cast<size_t>((size_t)info.uordblks);
   *resident_size += total_allocated_size;
 
   // Total allocated space is given by |uordblks|.
